presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/optimization/function/OperatorTranslators.java

```diff
@@ -33,36 +34,38 @@ private OperatorTranslators()
     }
 
     @ScalarOperator(ADD)
-    @SqlType(StandardTypes.BIGINT)
-    public static JdbcExpression add(@SqlType(StandardTypes.BIGINT) JdbcExpression left, @SqlType(StandardTypes.BIGINT) JdbcExpression right)
+    @SupportedSignatures({
+            @SqlSignature(argumentType = StandardTypes.BIGINT, returnType = StandardTypes.BIGINT),
+            @SqlSignature(argumentType = StandardTypes.INTEGER, returnType = StandardTypes.INTEGER)})
+    public static JdbcExpression add(JdbcExpression left, JdbcExpression right)
     {
         return new JdbcExpression(infixOperation("+", left, right), forwardBindVariables(left, right));
     }
 
     @ScalarOperator(SUBTRACT)
-    @SqlType(StandardTypes.BIGINT)
-    public static JdbcExpression subtract(@SqlType(StandardTypes.BIGINT) JdbcExpression left, @SqlType(StandardTypes.BIGINT) JdbcExpression right)
+    @SupportedSignatures(@SqlSignature(argumentType = StandardTypes.BIGINT, returnType = StandardTypes.BIGINT))
+    public static JdbcExpression subtract(JdbcExpression left, JdbcExpression right)
     {
         return new JdbcExpression(infixOperation("-", left, right), forwardBindVariables(left, right));
     }
 
     @ScalarOperator(EQUAL)
-    @SqlType(StandardTypes.BOOLEAN)
-    public static JdbcExpression equal(@SqlType(StandardTypes.BIGINT) JdbcExpression left, @SqlType(StandardTypes.BIGINT) JdbcExpression right)
+    @SupportedSignatures(@SqlSignature(argumentType = StandardTypes.BIGINT, returnType = StandardTypes.BOOLEAN))
```

Comment:
Since most JDBC support EQUALS (`=`), NOT_EQUALS (`<>`) , comparison functions (greater than, less than) out-of-the-box, lets add support for these standard type variants - `bigint, integer, smallint, tinyint, boolean, date, decimal, real, double, varchar, timestamp`